BlackStone8960/Musician-First
# Fix/signup with email

I fixed the bug Taichi found and mentioned [here](https://github.com/BlackStone8960/Musician-First/pull/130#issue-1187767973)

The bug details:
After users logged out and try to create another new account with their email without refreshing the app, and after you click on the 'save' button (image below), the app redirects them to `/filter1` page with the previous account information.
Then if they refresh the app, `/signup` page for the new account will be shown up.

Why this happened:
It was because redux, the state of userAccount still had the previous account info. `src/reducers/userAccount.js`.

How it has been fixed:
I basically added `deleteUserAccount()` action and set the state as {} after signing out `src/actions/auth.js`.

How to test:
1. Login with one of your accounts
2. Logout
3. Without refreshing the page, create a new account with an email address, not gmail
4. Fill in your name and password, and click the 'save' button (one on the image)
5. Right after clicking the 'save' button, see if the app redirects you to `/signup` page correctly and let you fill in the profile information

<img src="https://user-images.githubusercontent.com/69213541/161374990-f7d943e0-35ea-4bdb-8704-7f72601ff93d.jpg" width=300px>


Please test a few times to make sure the error doesn't occur, thank you!
